pest/src/lib.rs

```diff
@@ -6,8 +6,8 @@
 // license <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
 // option. All files in the project carrying such notice may not be copied,
 // modified, or distributed except according to those terms.
-
 #![cfg_attr(feature = "const_prec_climber", feature(const_fn))]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Personally I prefer `#[cfg(feature = "std")] extern crate std;` to `#![cfg_attr(not(feature = "std"), no_std)]`.

Using `extern crate std;` is more clearly additive, and while it means you don't get std prelude in std areas, it also means you don't get std prelude in no_std areas, so it's a lot harder to accidentally use std types, and easier to remember to import the alloc types.

Reply:
I believe you are trying to suggest to replace `#![cfg_attr(not(feature = "std"), no_std)]` by `#[cfg(feature = "std")] extern crate std;`. But this produces a compile error:

~~~
error[E0259]: the name `std` is defined multiple times
  --> pest/src/lib.rs:67:25
   |
67 | #[cfg(feature = "std")] extern crate std;
   |                         ^^^^^^^^^^^^^^^^^ `std` reimported here
   |
   = note: `std` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
67 | #[cfg(feature = "std")] extern crate std as other_std;
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
~~~

That is because the `std` crate is implicitly loaded by default, so we cannot conditionally include it. Instead, we need an explicit instruction to *disable* it. That's what `no_std` does. [Serde does it this way, for example.](https://github.com/serde-rs/serde/blob/5a8dcac2ed1407fab3f7fd23f2d56af42dcd448f/serde/src/lib.rs#L89)
I therefore think that there is no way around `#![cfg_attr(not(feature = "std"), no_std)]`.

Reply:
No, you can do both unconditional `#![no_std]` and `#[cfg] extern crate std;`.

Reply:
Ah, I understand! Thanks for pointing this out, I did not know about that before. I implemented your suggestion, see below.